### ISSUE TITLE
Distribute from root

### ISSUE THREAD

**Issue by thien-do** [thien-do/moai#31]
Traditionally a lib is built from "src" to "dist", then the whole project is published to npm. This means to use something via an absolute path we would do `import {} from "@moai/core/dist/something"`.

Previously, we think this is quite ugly and we do a little trick that we move the package.json to inside "dist" and then publish only the "dist" folder (as a whole project). This means the import above would now be nicer as `@moai/core/something`.

However, this yields more problems than benefits:
1. Is that we sym-link just won't work anymore. Yes we can sym-link to the dist folder itself, but most build will just delete the dist and create a new one instead of emptying its content.
2. Versioning is more difficult. We need to increase the version *before* the build because the "moving package.json" is a part of the build.

In conclusion, we should just rollback and publish from root

**Comment by kcjpop** [comment]
![image](https://user-images.githubusercontent.com/172503/106796571-36f94280-6664-11eb-8395-299915c353cc.png)

**Comment by thien-do** [comment]
Done
